packages/firestore/src/platform/platform.ts

```diff
@@ -50,6 +50,11 @@ export interface Platform {
    */
   randomBytes(nBytes: number): Uint8Array;
 
+  /**
+   * Builds a `ByteStreamReader` from a data source.
+   */
+  toByteStreamReader(source: unknown): ByteStreamReader;
```

Comment:
Can the source type be `Uint8Array | ArrayBuffer | ReadableStream<Uint8Array>`?

And since this type combination is used widely now, should we add a local type alias?